Handle unicode encode errors in changesets

--- site-packages/buildbot/status/web/feeds.py
+++ site-packages/buildbot/status/web/feeds.py
@@ -209,7 +209,11 @@
                             # Probably the log file has been removed
                             logdata ='** log file not available **'
                         unilist = list()
-                        unilist.append(unicode(line,'utf-8'))
+                        for line in logdata.split('\n')[-30:]:
+                            try:
+                                unilist.append(unicode(line,'utf-8'))
+                            except:
+                                unilist.append("*** bad unicode ***")
                         log_lines.extend(unilist)
 
             bc = {}

